influxdb_client/client/util/date_utils.py

```diff
@@ -78,12 +81,13 @@ def get_date_helper() -> DateHelper:
     If there is a 'ciso8601' than use 'ciso8601.parse_datetime' else use 'dateutil.parse'.
     """
     global date_helper
-    if date_helper is None:
-        date_helper = DateHelper()
-        try:
-            import ciso8601
-            date_helper.parse_date = ciso8601.parse_datetime
-        except ModuleNotFoundError:
-            date_helper.parse_date = parser.parse
+    with lock_:
+        if date_helper is None:
+            date_helper = DateHelper()
+            try:
+                import ciso8601
+                date_helper.parse_date = ciso8601.parse_datetime
+            except ModuleNotFoundError:
+                date_helper.parse_date = parser.parse
```

Comment:
The acquiring locks is an expensive operation. We can use something like:

```suggestion
    if date_helper is None:
        with lock_:
            # avoid duplicate initialization
            if date_helper is None:
                _date_helper = DateHelper()
                try:
                    import ciso8601
                    _date_helper.parse_date = ciso8601.parse_datetime
                except ModuleNotFoundError:
                    _date_helper.parse_date = parser.parse
                date_helper = _date_helper
```